File: AI_EXPORT_FOR_BLUEPRINT_HUB/frontend/chat.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, of, catchError } from 'rxjs';
import { map } from 'rxjs/operators';
import { PageType } from './chat.model'; // Adjust import path as needed

export interface ChatRequest {
  message: string;
  pageContext: string;
}

export interface ChatResponse {
  response: string;
  success: boolean;
}

@Injectable({
  providedIn: 'root',
})
export class ChatService {
  private apiUrl = '/api/chat';

  constructor(private http: HttpClient) {
    // TODO: Update with your Blueprint Hub backend URL
    // if (window.location.hostname.includes('your-domain.com') || 
    //     window.location.hostname === 'localhost') {
    //   this.apiUrl = 'https://your-blueprint-hub-backend.com/api/chat';
    // }
  }

  getContextualWelcome(pageType: PageType): string {
    switch (pageType) {
      case 'home':
        return 'Hi! I can help you navigate your Blueprint Hub dashboard, understand project management features, or explain how the blueprint system works. What would you like to know?';
      case 'projects':
        return 'Hello! I can help you with project management, filtering options, project creation, or any project-related features. How can I assist?';
      case 'blueprints':
        return 'Hi there! I can explain blueprint templates, categories, usage patterns, or help you find the right blueprint for your needs. What questions do you have?';
      case 'detail':
        return 'Welcome! I can help you understand the details on this page, explain specific features, or guide you through available actions. What would you like to know?';
      case 'settings':
        return 'Hi! I can help you configure your Blueprint Hub settings, manage integrations, or explain customization options. What would you like to know?';
      default:
        return "Hello! I'm here to help you with any questions about this page. What can I assist you with?";
    }
  }

  generateAIResponse(
    userMessage: string,
    pageType: PageType
  ): Observable<string> {
    const request: ChatRequest = {
      message: userMessage,
      pageContext: pageType
    };

    console.log('Chat service making request to:', `${this.apiUrl}/message`);
    console.log('Request payload:', request);

    return this.http.post<ChatResponse>(`${this.apiUrl}/message`, request).pipe(
      map(response => {
        console.log('Chat API response received:', response);
        return response.response;
      }),
      catchError(error => {
        console.error('Chat API error:', error);
        console.log('Falling back to local response');
        // Fallback to local response if API fails
        return of(this.getContextualResponse(userMessage, pageType));
      })
    );
  }

  private getContextualResponse(
    userMessage: string,
    pageType: PageType
  ): string {
    const lowerMessage = userMessage.toLowerCase();

    // Context-aware responses based on page type and user input
    if (pageType === 'home') {
      if (lowerMessage.includes('search') || lowerMessage.includes('find')) {
        return 'You can search for projects and blueprints using the search bar. It searches through project names, descriptions, tags, and blueprint content. Use filters to narrow down results by category, status, or date.';
      }
      if (lowerMessage.includes('project') || lowerMessage.includes('create')) {
        return 'To create a new project, click the "New Project" button. You can start from scratch or use one of our blueprint templates to accelerate your setup with pre-configured structures and best practices.';
      }
      if (lowerMessage.includes('blueprint')) {
        return 'Blueprints are reusable templates that help you quickly set up projects with proven architectures and configurations. Browse the blueprint library to find templates for your technology stack.';
      }
    }

    if (pageType === 'projects') {
      if (lowerMessage.includes('status') || lowerMessage.includes('filter')) {
        return 'Projects can be filtered by status (Active, Completed, Archived), technology stack, team members, or creation date. Use the filter panel to narrow down your project list.';
      }
      if (lowerMessage.includes('collaborate') || lowerMessage.includes('team')) {
        return 'Projects support team collaboration with role-based access, shared resources, and integrated communication tools. Invite team members and assign appropriate permissions.';
      }
    }

    if (pageType === 'blueprints') {
      if (lowerMessage.includes('category') || lowerMessage.includes('type')) {
        return 'Blueprints are organized by categories like Web Development, Mobile Apps, DevOps, Data Science, etc. Each blueprint includes documentation, sample code, and configuration files.';
      }
      if (lowerMessage.includes('customize') || lowerMessage.includes('modify')) {
        return 'You can customize blueprints by forking them to create your own versions, or by using them as starting points and modifying the generated project structure.';
      }
    }

    if (pageType === 'detail') {
      if (lowerMessage.includes('deploy') || lowerMessage.includes('build')) {
        return 'Use the deployment section to configure CI/CD pipelines, set up environments, and manage builds. Each project can have multiple deployment targets.';
      }
      if (lowerMessage.includes('resource') || lowerMessage.includes('file')) {
        return 'Project resources include source code, documentation, configuration files, and shared assets. Use the file browser to navigate and manage your project structure.';
      }
    }

    if (pageType === 'settings') {
      if (lowerMessage.includes('integration') || lowerMessage.includes('api')) {
        return 'Configure integrations with version control systems, cloud providers, and development tools. Set up API keys and authentication for seamless workflows.';
      }
      if (lowerMessage.includes('notification') || lowerMessage.includes('alert')) {
        return 'Customize notification preferences for project updates, build status, team activities, and system alerts. Choose email, in-app, or webhook delivery methods.';
      }
    }

    // General responses
    if (lowerMessage.includes('help') || lowerMessage.includes('how')) {
      return "I can help explain any feature you see on this page. Try asking about specific functionality, workflows, or best practices for using Blueprint Hub effectively.";
    }

    // Default response with context
    return `I understand you're asking about "${userMessage}". Based on this ${pageType} page, I'd be happy to help explain specific features or guide you through available options. Could you be more specific about what you'd like to know?`;
  }
}

File: AI_EXPORT_FOR_BLUEPRINT_HUB/backend/ChatService.java
package your.blueprint.hub.service; // UPDATE: Change to your package

import com.azure.ai.openai.OpenAIClient;
import com.azure.ai.openai.OpenAIClientBuilder;
import com.azure.ai.openai.models.*;
import com.azure.core.credential.AzureKeyCredential;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import java.util.ArrayList;
import java.util.List;

@Service
public class ChatService {
    
    private final OpenAIClient client;
    private final String deploymentName;
    
    public ChatService(
            @Value("${azure.openai.endpoint}") String endpoint,
            @Value("${azure.openai.api-key}") String apiKey,
            @Value("${azure.openai.deployment-name}") String deploymentName) {
        
        this.deploymentName = deploymentName;
        this.client = initializeClient(endpoint, apiKey, deploymentName);
    }
    
    private OpenAIClient initializeClient(String endpoint, String apiKey, String deploymentName) {
        try {
            // Check if all required configuration is present
            if (endpoint == null || endpoint.equals("https://your-openai.openai.azure.com/") || 
                apiKey == null || apiKey.equals("your-api-key") ||
                deploymentName == null || deploymentName.equals("gpt-4")) {
                
                System.out.println("Azure OpenAI not properly configured, using fallback mode");
                return null;
            }
            
            OpenAIClient openAIClient = new OpenAIClientBuilder()
                    .endpoint(endpoint)
                    .credential(new AzureKeyCredential(apiKey))
                    .buildClient();
                    
            System.out.println("Azure OpenAI client initialized successfully");
            return openAIClient;
        } catch (Exception e) {
            System.err.println("Failed to initialize Azure OpenAI client: " + e.getMessage());
            return null;
        }
    }
    
    public String generateResponse(String userMessage, String pageContext) {
        try {
            // If OpenAI client is not available, use fallback response
            if (client == null) {
                return getFallbackResponse(userMessage, pageContext);
            }
            
            List<ChatRequestMessage> chatMessages = new ArrayList<>();
            
            // Add system message with context
            String systemPrompt = buildSystemPrompt(pageContext);
            chatMessages.add(new ChatRequestSystemMessage(systemPrompt));
            
            // Add user message
            chatMessages.add(new ChatRequestUserMessage(userMessage));
            
            ChatCompletionsOptions chatCompletionsOptions = new ChatCompletionsOptions(chatMessages);
            chatCompletionsOptions.setMaxTokens(500);
            chatCompletionsOptions.setTemperature(0.7);
            
            ChatCompletions chatCompletions = client.getChatCompletions(deploymentName, chatCompletionsOptions);
            
            if (chatCompletions.getChoices() != null && !chatCompletions.getChoices().isEmpty()) {
                return chatCompletions.getChoices().get(0).getMessage().getContent();
            }
            
            return "I'm sorry, I couldn't generate a response at the moment. Please try again.";
            
        } catch (Exception e) {
            System.err.println("Error calling Azure OpenAI: " + e.getMessage());
            return getFallbackResponse(userMessage, pageContext);
        }
    }
    
    private String getFallbackResponse(String userMessage, String pageContext) {
        String lowerMessage = userMessage.toLowerCase();
        
        // Blueprint Hub specific fallback responses
        if (lowerMessage.contains("project") || lowerMessage.contains("create")) {
            return "To create a new project, click the 'New Project' button. You can start from scratch or use one of our blueprint templates to accelerate development with proven architectures.";
        }
        
        if (lowerMessage.contains("blueprint") || lowerMessage.contains("template")) {
            return "Blueprints are reusable project templates that include pre-configured code, documentation, and best practices. Browse the blueprint library to find templates for your technology stack.";
        }
        
        if (lowerMessage.contains("search") || lowerMessage.contains("find")) {
            return "You can search for projects and blueprints using the search bar. It searches through project names, descriptions, tags, and blueprint content.";
        }
        
        if (lowerMessage.contains("deploy") || lowerMessage.contains("build")) {
            return "Use the deployment section to set up CI/CD pipelines, configure environments, and manage builds. Each project can have multiple deployment targets.";
        }
        
        if (lowerMessage.contains("collaborate") || lowerMessage.contains("team")) {
            return "Projects support team collaboration with role-based access, shared resources, and integrated communication tools. Invite team members and assign appropriate permissions.";
        }
        
        return "Hello! I'm here to help you with Blueprint Hub. You can ask me about creating projects, using blueprints, deploying applications, team collaboration, or any other features you'd like to learn about.";
    }
    
    private String buildSystemPrompt(String pageContext) {
        String basePrompt = "You are an AI assistant for Blueprint Hub, a platform for managing software projects and reusable blueprints. " +
                "You help users with project management, blueprint usage, deployment, collaboration, and platform navigation. " +
                "Be helpful, concise, and professional in your responses.";
        
        switch (pageContext.toLowerCase()) {
            case "home":
            case "dashboard":
                return basePrompt + " The user is currently on the main dashboard where they can see " +
                        "their projects overview, recent activity, and quick actions. Help them understand " +
                        "dashboard features, project summaries, and navigation options.";
                        
            case "projects":
                return basePrompt + " The user is on the projects page where they can view, search, " +
                        "and filter their projects. Help them with project management, filtering options, " +
                        "project status, and team collaboration features.";
                        
            case "blueprints":
                return basePrompt + " The user is browsing the blueprint library where they can find " +
                        "and use project templates. Help them understand blueprint categories, " +
                        "customization options, and how to apply blueprints to new projects.";
                        
            case "detail":
                return basePrompt + " The user is viewing a specific project's details page. Help them " +
                        "understand project information, resources, deployment options, team members, " +
                        "and available actions for this project.";
                        
            case "settings":
                return basePrompt + " The user is on the settings page where they can configure their " +
                        "preferences, integrations, and account settings. Help them with configuration " +
                        "options, API integrations, and feature explanations.";
                        
            default:
                return basePrompt + " Provide general help about the Blueprint Hub platform.";
        }
    }
}